Move PTL create_vnodes method from class Server to Mom

visheshh opened the pull request:
<!--- Please review your changes in preview mode -->
<!--- Provide a general summary of your changes in the Title above -->

#### Describe Bug or Feature
Move PTL create_vnodes method from class Server to Mom


#### Describe Your Change
1.Moved PTL create_vnodes method from class Server to Mom
2. Removed the mom object and the vnode_prefix name sent to create_vnodes function and added back an optional vnode_prefix which can be given under necessary conditions.
3.Make mom hostname as the default vnode_prefix and removed the vnode_prefix from tests if not absolutely necessary.
4. Modified TestOfflineVnode test suite to verify vnodes from both the moms.

#### Link to Design Doc
<!--- If there is a design, link to it here: **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)** -->
https://openpbs.atlassian.net/wiki/spaces/PD/pages/2083028993/Move+create+vnodes+from+Class+Server+to+Mom+in+PTL

#### Attach Test and Valgrind Logs/Output
<!--- Please attach your test log output from running the test you added (or from existing tests that cover your changes) -->
<!--- Don't forget to run Valgrind if appropriate and attach the resulting logs -->
Mainline :
Description: Tests from given sources on platforms UBUNTU1804, SUSE15, CENTOS8, CENTOS7
Platforms: UBUNTU1804,SUSE15,CENTOS8,CENTOS7
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|3759|968|3|2|1|1|961|

create_vnodes branch:
Description: Tests from given sources on platforms UBUNTU1804, SUSE15, CENTOS8, CENTOS7
Platforms: UBUNTU1804,SUSE15,CENTOS8,CENTOS7
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|3758|968|3|1|1|1|962|


<!--- Pull Request Guidelines: [Pull Request Guidelines](https://pbspro.atlassian.net/wiki/spaces/DG/pages/1187348483/Pull+Request+Guidelines) -->

bhroam commented on test/tests/functional/pbs_test_tpp.py:
```diff
@@ -354,10 +354,12 @@ def test_comm_with_vnode_insertion(self):
-            vnode_val = "vnode=vn[0]:ncpus=1+vnode=vn[1]:ncpus=1"
+            self.mom.create_vnodes(a, 2)
+            vnode_val = "vnode=" + vn + \
+                "[0]:ncpus=1+vnode=" + vn + "[1]:ncpus=1"
```
Consider using string formatting.  If you do that, you might be able to put the string on the first line and the (vn, vn) on the second.

bhroam commented on test/tests/functional/pbs_test_tpp.py:
```diff
@@ -374,7 +376,7 @@ def test_comm_with_vnode_insertion(self):
             vnode_list = [self.server.status(NODE)[1]['id'],
                           self.server.status(NODE)[2]['id']]
```
Only call status() once, and and call ret[0]['id'] and ret[1]['id]], or use list comprehension and say [ret[i]['id'] for i in range(2)] (here and above)

agrawalravi90 reviewed (dismissed):
"Move PTL create_vnodes method from class Server to Mom"

Can you please explain the motivation to do this?

visheshh commented:
> 
> 
> "Move PTL create_vnodes method from class Server to Mom"
> 
> Can you please explain the motivation to do this?

Hi @agrawalravi90 ,
Motivation to do this was to make vnode prefixes same across all platforms such as cpuset moms.
We are removing skipcpuset decorator and trying to return available vnodes in the system which have mom name as prefix and tests use vn or vnode as prefix, so we removed them and in the process moved create_vnodes function which has nothing do with server object and instead of calling from server object and passing mom object as argument we moved it under class mom.

visheshh commented:
Thanks @bhroam for taking your time and doing a quick review for all of these tests.

visheshh commented:
Hi @bhroam, I have addressed all comments. Please review.

visheshh commented:
@bhroam Please review

bhroam reviewed (approved):
LGTM

arungrover reviewed (approved):
LGTM
